File: app/src/main/java/com/mdkashif/universalarm/utils/persistence/AppDatabase.java
package com.mdkashif.universalarm.utils.persistence;

import android.content.Context;

import com.mdkashif.universalarm.alarm.miscellaneous.db.DaoAccess;
import com.mdkashif.universalarm.alarm.miscellaneous.db.DaysModel;
import com.mdkashif.universalarm.alarm.miscellaneous.db.TimingsModel;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

@Database(entities = {TimingsModel.class, DaysModel.class}, version = 1)
public abstract class AppDatabase extends RoomDatabase {

    private static AppDatabase mINSTANCE;

    public abstract DaoAccess userDao();

    public static AppDatabase getAppDatabase(Context context) {
        if (mINSTANCE == null) {
            mINSTANCE =
                    Room.databaseBuilder(context.getApplicationContext(), AppDatabase.class, "universal-alarm-db")
                            .allowMainThreadQueries()
                            .build();
        }
        return mINSTANCE;
    }

    public static void destroyInstance() {
        mINSTANCE = null;
    }
}

File: app/src/main/java/com/mdkashif/universalarm/utils/persistence/AppDatabase.kt
package com.mdkashif.universalarm.utils.persistence

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import com.mdkashif.universalarm.alarm.miscellaneous.model.DaysModel
import com.mdkashif.universalarm.alarm.miscellaneous.model.LocationsModel
import com.mdkashif.universalarm.alarm.miscellaneous.model.TimingsModel

@Database(entities = [TimingsModel::class, DaysModel::class, LocationsModel::class], version = 1)
abstract class AppDatabase : RoomDatabase() {
    abstract fun accessDao(): RoomAccessDao

    companion object {
        private var mINSTANCE: AppDatabase? = null

        fun getAppDatabase(context: Context): AppDatabase {
            if (mINSTANCE == null) {
                mINSTANCE = Room.databaseBuilder(context.applicationContext, AppDatabase::class.java, "universal-alarm-db")
                        .build()
            }
            return mINSTANCE!!
        }
    }
}
